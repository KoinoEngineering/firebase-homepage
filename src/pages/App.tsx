import { createStyles, makeStyles } from "@material-ui/core";
import React from "react";
import { Redirect, Route as PublicRoute, BrowserRouter as Router, Switch } from "react-router-dom";
import MainFrame from "src/components/templates/MainFrame/MainFrame";
import ROUTES from "src/utils/routes";
import Top from "./Top";
import Sorts from "./Sorts";

const useStyles = makeStyles(
    createStyles({
        root: { height: "100%" },
    }),
);

const App: React.FC = () => {
    const classes = useStyles();
    return (
        <MainFrame>
            <div id="App" className={classes.root}>
                <Router>
                    <Switch>
                        <Redirect exact from="/" to={ROUTES.TOP} />
                        <PublicRoute exact path={ROUTES.TOP}>
                            <Top />
                        </PublicRoute>
                        <PublicRoute exact path={ROUTES.SORTS}>
                            <Sorts />
                        </PublicRoute>
                        <Redirect exact from="*" to={ROUTES.TOP} />
                    </Switch>
                </Router>
            </div>
        </MainFrame>
    );
};
export default App;
