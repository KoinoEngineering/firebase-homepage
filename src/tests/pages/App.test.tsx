import React from "react";
import App from "src/pages/App";
import { renderWithProvider } from "../tetutils";

jest.mock("src/components/templates/Profile", () => () => <div>Profile</div>);

test("App expect to match snapshot", () => {
    const target = renderWithProvider(<App />);
    expect(target).toMatchSnapshot();
});
