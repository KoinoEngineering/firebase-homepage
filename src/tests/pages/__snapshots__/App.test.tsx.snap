// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App expect to match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="MainFrame"
      >
        <header
          class="MuiPaper-root MuiAppBar-root makeStyles-root-3 MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
          id="Header"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
            >
              <h1
                class="MuiTypography-root MuiTypography-h1"
              >
                こいのエンジニアリングのホームページ
              </h1>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <a
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root makeStyles-root-4"
                href="https://github.com/KoinoEngineering/firebase-homepage"
                rel="noopener noreferrer"
                tabindex="0"
                target="_blank"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <div
                    class="MuiAvatar-root MuiAvatar-circle"
                  >
                    <img
                      alt="GitHub repository"
                      class="MuiAvatar-img"
                      src="GitHub-Mark-Light-64px.png"
                    />
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
        </header>
        <div
          class="makeStyles-main-2"
          id="main"
        >
          <div
            class="makeStyles-root-1"
            id="App"
          >
            <div
              class="MuiContainer-root TopPageContainer-root-207 MuiContainer-maxWidthLg"
              id="Top"
            >
              ホームページトップ
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="MainFrame"
    >
      <header
        class="MuiPaper-root MuiAppBar-root makeStyles-root-3 MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
        id="Header"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
          >
            <h1
              class="MuiTypography-root MuiTypography-h1"
            >
              こいのエンジニアリングのホームページ
            </h1>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root makeStyles-root-4"
              href="https://github.com/KoinoEngineering/firebase-homepage"
              rel="noopener noreferrer"
              tabindex="0"
              target="_blank"
            >
              <span
                class="MuiIconButton-label"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circle"
                >
                  <img
                    alt="GitHub repository"
                    class="MuiAvatar-img"
                    src="GitHub-Mark-Light-64px.png"
                  />
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </header>
      <div
        class="makeStyles-main-2"
        id="main"
      >
        <div
          class="makeStyles-root-1"
          id="App"
        >
          <div
            class="MuiContainer-root TopPageContainer-root-207 MuiContainer-maxWidthLg"
            id="Top"
          >
            ホームページトップ
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
